addons/web/static/tests/views/kanban_view_tests.js

```diff
@@ -11809,4 +11809,68 @@ QUnit.module("Views", (hooks) => {
         assert.deepEqual(getCardTexts(0), ["blip-4"]);
         assert.deepEqual(getCardTexts(1), ["yop10", "blip9", "gnap17", "cba13"]);
     });
+
+    QUnit.test("no demo data when all groups are folded then one is unfolded", async (assert) => {
+        await makeView({
+            type: "kanban",
+            resModel: "partner",
+            serverData,
+            arch: /* xml */ `
+                <kanban sample="1">
+                    <templates>
+                        <div t-name="kanban-box">
+                            <field name="id"/>
+                        </div>
+                    </templates>
+                </kanban>`,
+            groupBy: ["product_id"],
+            async mockRPC(route, args, performRPC) {
+                if (args.method === "web_read_group") {
+                    const result = await performRPC(route, args);
+                    for (const group of result.groups) {
+                        group.__fold = true;
+                    }
+                    return result;
+                }
+                return performRPC(route, args);
+            },
+        });
+
+        const groupHandle = target.querySelector(".o_kanban_group");
+        await click(groupHandle);
+
+        assert.containsN(target, ".o_kanban_record", 2); // instead of the the many in the demo data
+    });
+
+    QUnit.test(
+        "no content helper when all groups are folded but there are (unloaded) records",
+        async (assert) => {
+            await makeView({
+                type: "kanban",
+                resModel: "partner",
+                serverData,
+                arch: /* xml */ `
+                <kanban>
+                    <templates>
+                        <div t-name="kanban-box">
+                            <field name="id"/>
+                        </div>
+                    </templates>
+                </kanban>`,
+                groupBy: ["product_id"],
+                async mockRPC(route, args, performRPC) {
+                    if (args.method === "web_read_group") {
+                        const result = await performRPC(route, args);
+                        for (const group of result.groups) {
+                            group.__fold = true;
+                        }
+                        return result;
+                    }
+                    return performRPC(route, args);
+                },
+            });
+
```

Comment:
maybe assert the count in folded groups (it's written in the interface)

Reply:
@aab-odoo Not sure it has anything to do with this test. We just want to make sure there is no content helper

Reply:
actually, I withdraw what I said